File: client/src/utils/crypto/RSA.js
import JSEncrypt from 'jsencrypt'

/**
 * Encrypts data with RSA public key
 * @param data data to encrypt
 * @param publicKey RSA public key
 * @returns {string | false} encrypted data or false if error
 */
const encrypt = (data,publicKey) => {
  const encrypt = new JSEncrypt()
  encrypt.setPublicKey(publicKey)
    console.log(encrypt.encrypt(data))
  return encrypt.encrypt(data)
}

/**
 * Decrypts data with RSA private key
 * @param data data to decrypt
 * @param privateKey RSA private key
 * @returns {string | false} decrypted data or false if error
 */
const decrypt = (data,privateKey) => {
  const decrypt = new JSEncrypt()
  decrypt.setPrivateKey(privateKey)
  return decrypt.decrypt(data)
}

module.exports = {
  encrypt,
  decrypt
}

File: serve/utils/crypto/RSA.py
from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_v1_5
import base64


def encrypt(data, public_key):
    """
    Encrypts data with RSA public key
    :param data: data to encrypt
    :param public_key: public key
    :return: encrypted data
    """
    cipher = PKCS1_v1_5.new(RSA.importKey(public_key))
    cipher_text = base64.b64encode(cipher.encrypt(data.encode()))
    return cipher_text.decode()


def decrypt(cipher_text, private_key):
    """
    Decrypts data with RSA private key
    :param cipher_text: data to decrypt
    :param private_key: private key
    :return: decrypted data
    """
    cipher = PKCS1_v1_5.new(RSA.importKey(private_key))
    message = cipher.decrypt(base64.b64decode(cipher_text), 'ERROR')
    return message.decode()
